bin/portal-bridge/src/api/consensus/mod.rs

```diff
@@ -0,0 +1,210 @@
+            "/eth/v1/beacon/light_client/updates?start_period={start_period}&count={count}"
+        );
+        Ok(self.request(endpoint, None).await?.data)
+    }
+
+    /// Requests the latest `LightClientOptimisticUpdate` known by the server.
+    pub async fn get_light_client_optimistic_update(
+        &self,
+    ) -> anyhow::Result<LightClientOptimisticUpdateDeneb> {
+        let endpoint = "/eth/v1/beacon/light_client/optimistic_update".to_string();
+        Ok(self.request(endpoint, None).await?.data)
+    }
+
+    /// Requests the latest `LightClientFinalityUpdate` known by the server.
+    pub async fn get_light_client_finality_update(
+        &self,
+    ) -> anyhow::Result<LightClientFinalityUpdateDeneb> {
+        let endpoint = "/eth/v1/beacon/light_client/finality_update".to_string();
+        Ok(self.request(endpoint, None).await?.data)
+    }
+
+    /// Requests the Node's `Version` string.
+    pub async fn get_node_version(&self) -> anyhow::Result<VersionResponse> {
+        let endpoint = "/eth/v1/node/version".to_string();
+        Ok(self.request(endpoint, None).await?.data)
+    }
+
+    /// Requests the `BeaconState` structure corresponding to the current head of the beacon chain.
+    pub async fn get_beacon_state(&self) -> anyhow::Result<BeaconStateDeneb> {
+        let endpoint = "/eth/v2/debug/beacon/states/finalized".to_string();
+        Ok(self
+            .request(endpoint, Some(DEFAULT_BEACON_STATE_REQUEST_TIMEOUT))
+            .await?
+            .data)
+    }
+
+    /// Make a request to the cl provider. If the primary provider fails, it will retry with the
+    /// fallback.
+    async fn request<T>(
+        &self,
+        endpoint: String,
+        custom_timeout: Option<Duration>,
+    ) -> anyhow::Result<VersionedDataResponse<T>>
+    where
+        T: Decode + DeserializeOwned,
+    {
+        match Self::request_no_fallback(endpoint.clone(), &self.primary, custom_timeout).await {
+            Ok(response) => Ok(response),
+            Err(err) => {
+                warn!("Error requesting consensus data from provider, retrying with fallback provider: {err:?}");
+                Self::request_no_fallback(endpoint, &self.fallback, custom_timeout)
+                    .await
+                    .map_err(|err| {
+                        anyhow!("Unable to request consensus data from fallback provider: {err:?}")
+                    })
+            }
+        }
+    }
+
+    /// request() should be used instead of this function.
+    async fn request_no_fallback<T>(
+        endpoint: String,
+        client: &ClientWithBaseUrl,
+        custom_timeout: Option<Duration>,
+    ) -> anyhow::Result<VersionedDataResponse<T>>
+    where
+        T: Decode + DeserializeOwned,
+    {
+        let request = client.get(&endpoint)?;
+        let request = if let Some(timeout) = custom_timeout {
+            request.timeout(timeout)
+        } else {
+            request
+        };
```

Comment:
```suggestion
        let request = match custom_timeout {
            Some(timeout) => request.timeout(timeout),
            None => request,
        };
```

tiniest nitpick: I found suggested code a bit more readable. up to you